libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -217,12 +217,13 @@ const AP_Param::GroupInfo AP_OSD::var_info[] = {
     AP_SUBGROUPINFO(param_screen[1], "6_", 22, AP_OSD, AP_OSD_ParamScreen),
 #endif
 
+#if OSD_ENABLED
```

Comment:
Why do we need this ?

Reply:
[This](https://github.com/ArduPilot/ardupilot/blob/75b9a3ff7767e0bc78c6e8ca8f5b9b983c7c1b15/libraries/AP_OSD/AP_OSD.h#L512) line is compiled out when OSD is disabled. And if we dont have this line, we can't use `screen` array here. So compiled that out too.